src/utils/npm.js

```diff
@@ -1,18 +1,50 @@
 // @flow
-import * as processes from './processes';
+import {PError} from './errors';
 import * as logger from './logger';
+import * as processes from './processes';
 import pLimit from 'p-limit';
 
-const npmRequestLimit = pLimit(20);
+const npmRequestLimit = pLimit(40);
 
 export function info(pkgName: string) {
   return npmRequestLimit(async () => {
     logger.info(`npm info ${pkgName}`);
 
-    let result = await processes.spawn('npm', ['info', pkgName, '--json'], {
+    const result = await processes.spawn('npm', ['info', pkgName, '--json'], {
       silent: true,
     });
 
     return JSON.parse(result.stdout);
   });
 }
+
+export function addTag(pkgName: string, pkgVersion: string, tag: string) {
+  return npmRequestLimit(async () => {
+    const pkgStr = `${pkgName}@${pkgVersion}`;
+    logger.info(`npm dist-tag ${pkgStr} ${tag}`);
+
+    const result = await processes.spawn('npm', ['dist-tag', 'add', pkgStr, tag]);
+    // An existing tag will return a warning to stderr, but a 0 status code
+    if (result.stderr) {
+      throw new PError(`Could not lock ${pkgStr} as a lock already exists`);
+    }
+    return result;
+  });}
+
+
+export async function removeTag(pkgName: string, tag: string) {
+  return npmRequestLimit(async () => {
+    logger.info(`npm dist-tag rm ${pkgName} ${tag}`);
+
+    try {
+      return await processes.spawn('npm', ['dist-tag', 'rm', pkgName, tag]);
+    } catch (error) {
+      // The dist tag not existing is unexpected, but shouldn't prevent execution
```

Comment:
Thoughts on this?
It's bad that we are using an error message as "api", but better than halting because you expected a tag?

Reply:
Should we be adding this logic in this function?

You could always have:

```js
npm.removeTag(); // throws a nice error when there's no dist-tag
npm.removeTagSafe(); // does not throw an error, logs
```

Reply:
Sure, we can. Which would you want to call in this case though?